app/downloadshandler.py

```diff
@@ -0,0 +1,122 @@
+downloads_routes = RouteTableDef()
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
+
+STATIC_DIR = os.path.abspath('../static')
+
+
+def setup_request(request):
+    request['client_ip'] = request.headers.get('X-Forwarded-For', None)
+
+
+def log_entry(request, endpoint):
+    method = request.method
+    logger.info(f"received {method} on endpoint '{endpoint}'",
+                method=request.method,
+                path=request.path)
+
+
+@downloads_routes.view('/downloads/{download_type}')
+class DownloadsPage:
+    @aiohttp_jinja2.template('downloads.html')
+    async def get(self, request):
+        download_type = request.match_info['download_type']
+        session = await get_session(request)
+        await saml.ensure_logged_in(request)
+
+        user_role = await saml.get_role_id(request)
+        
+        if download_permission(user_role):
+            pass 
+        else:
+            return aiohttp_jinja2.render_template('error404.html', request,
+                    {'include_nav': True})
+
+        try:
+            search_range, records_per_page = page_bounds(1)
+
+            get_employee_info = get_employee_records(search_range, iat = True)
+            get_employee_info_json = get_employee_info.json() 
+
+            if len(get_employee_info_json) > 0: 
+                employee_sum = get_employee_info_json[0].get('total_employees',0) 
+
+                # If there are more than 0 people in iat
+                # then get_employee_info is set to everyone (max is total number of employees)
+                search_range = {'rangeHigh': employee_sum, 'rangeLow': 0}
+                get_employee_info = get_employee_records(search_range, iat = True)
+                get_employee_info_json = get_employee_info.json() 
+
+                html_employee_records = iat_employee_record_table(get_employee_info_json, remove_html=True)
+                html_headers = iat_employee_table_headers()
+
+        except ClientResponseError as ex:
+            if ex.status == 503:
+                ip = request['client_ip']
+                logger.warn('Server is unavailable', client_ip=ip)
+                flash(request, SERVICE_DOWN_MSG)
+                return aiohttp_jinja2.render_template('error503.html', request,
+                                                      {'include_nav': False})
+            else:
+                raise ex
+
+        if get_employee_info.status_code == 200:
+
+            headers = "" 
+            for html_header in html_headers:
+                headers = headers + str(html_header.get('value')) + " , "
+            headers = headers[:-3]
+
+            rows=""
+            for employee_record  in  html_employee_records:
+                rows = str(rows) + "\n"
+                for each_array in employee_record.get('tds'):
+                    rows = str(rows) + str(each_array.get('value')) + " , "
+                rows = rows[:-3]
+            
+            if download_type == "iat":
```

Comment:
There doesn't seem to be any handling for when the download_type is a non-standard value. Is this intended, or should there be an else here to catch unknowns and error?

Reply:
Ahh, this is just  for the future if more  download   types are required, something that might be required later on. The only change would need to be  to add the button and a case for it in this statement (currently the function is never run without "iat")

Reply:
It's more that there should probably at least be a warning emitted into the logs, if the wrong value is passed in from a different part of the UI, rather than silently doing nothing.